plugins/utils/flash.nix

```diff
@@ -6,536 +6,670 @@
-                }
-              ''
-              "options used when flash is activated through a regular search with `/` or `?`";
-          treesitter =
-            mkModeConfig { }
-              ''
-                {
-                  labels = "abcdefghijklmnopqrstuvwxyz";
-                  jump = { pos = "range"; };
-                  search = { incremental = false; };
-                  label = { before = true; after = true; style = "inline"; };
-                  highlight = { backdrop = false; matches = false; };
-                }helpers.ifNonNull'
-              ''
-              ''
-                options used for treesitter selections `require("flash").treesitter()`
-              '';
-          treesitterSearch = mkModeConfig { } ''
-            {
-              jump = { pos = "range"; };
-              search = { multiWindow = true; wrap = true; incremental = false; };
-              remoteOp = { restore = true };
-              label = { before = true; after = true; style = "inline"; };
-            }
-          '' "";
-          remote = mkModeConfig { } ''
-            { remoteOp = { restore = true; motion = true; }; }
-          '' "options used for remote flash";
-        };
-    }
-    // configOpts;
+            keys = helpers.defaultNullOpts.mkAttrsOf' {
+              type = types.str;
+              pluginDefault = defaults.keys;
+              description = ''
+                By default all keymaps are enabled, but you can disable some of them,
+                by removing them from the list.
 
-  config =
-    let
-      mkGlobalConfig = c: {
-        inherit (c) labels;
-        search =
-          if c.search.automatic then
-            helpers.emptyTable
-          else
-            {
-              multi_window = c.search.multiWindow;
-              inherit (c.search)
-                forward
-                wrap
-                mode
-                incremental
-                exclude
-                trigger
-                ;
-              max_length = c.search.maxLength;
+                If you rather use another key, you can map them to something else.
+              '';
+              example = {
+                ";" = "L";
+                "," = "H";
+              };
             };
-        jump = {
-          inherit (c.jump)
-            jumplist
-            pos
-            history
-            register
-            nohlsearch
-            autojump
-            inclusive
-            offset
-            ;
-        };
-        label = {
-          inherit (c.label)
-            uppercase
-            exclude
-            current
-            after
-            before
-            style
-            reuse
-            distance
-            ;
-          min_pattern_length = c.label.minPatternLength;
-          rainbow = {
-            inherit (c.label.rainbow) enabled shade;
+
+            char_actions = helpers.defaultNullOpts.mkLuaFn defaults.char_actions ''
```

Comment:
Think this type is wrong? Upstream docs say:

> alias Flash.CharActions table<string, "next" | "prev" | "right" | "left">
> The direction for `prev` and `next` is determined by the motion.
> `left` and `right` are always left and right.

Reply:
I guess it is right, the [only usage](https://github.com/folke/flash.nvim/blob/43f67935d388fbb540f8b40e8cbfd80de54f978a/lua/flash/plugins/char.lua#L200) calls it as a function.